pkg/disttask/framework/taskexecutor/task_executor_test.go

```diff
@@ -526,3 +527,159 @@ func TestCheckBalanceSubtask(t *testing.T) {
 	taskExecutor.checkBalanceSubtask(ctx)
 	require.True(t, ctrl.Satisfied())
 }
+
+func TestExecutorErrHandling(t *testing.T) {
+	var tp proto.TaskType = "test_task_executor"
+	var taskID int64 = 1
+	var concurrency = 10
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	runCtx, runCancel := context.WithCancel(ctx)
+	defer runCancel()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockSubtaskTable := mock.NewMockTaskTable(ctrl)
+	mockSubtaskExecutor := mockexecute.NewMockSubtaskExecutor(ctrl)
+	mockExtension := mock.NewMockExtension(ctrl)
+	task := &proto.Task{Step: proto.StepOne, Type: tp}
+	taskExecutor := NewBaseTaskExecutor(ctx, "id", task, mockSubtaskTable)
+	taskExecutor.Extension = mockExtension
+
+	// 1. GetSubtaskExecutor meet retryable error.
+	getSubtaskExecutorErr := errors.New("get executor err")
+	mockExtension.EXPECT().GetSubtaskExecutor(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, getSubtaskExecutorErr)
+	mockExtension.EXPECT().IsRetryableError(gomock.Any()).Return(true)
+	require.NoError(t, taskExecutor.Run(runCtx, &proto.Task{Step: proto.StepOne, Type: tp, ID: taskID, Concurrency: concurrency}))
+	require.True(t, ctrl.Satisfied())
```

Comment:
error return from GetSubtaskExecutor should be fatal that will not be retried(we need comment this in the api definition)